code/game/objects/items/food/burgers.dm

```diff
@@ -297,6 +297,14 @@
 	foodtypes = GRAIN | MEAT | DAIRY
 	venue_value = FOOD_PRICE_EXOTIC
 
+/obj/item/food/burger/superbite/suicide_act(mob/user, show_message=TRUE)
+	. = OXYLOSS
+	user.visible_message(span_suicide("[user] starts to eat [src] in one bite, it looks like [user.p_theyre()] trying to commit suicide!"))
+	var/datum/component/edible/C = GetComponent(/datum/component/edible)
```

Comment:
https://github.com/tgstation/tgstation/blob/master/.github/guides/STYLE.md#use-descriptive-and-obvious-names

Reply:
Not really happy about this GetComponent either but I don't know another way you'd do this cleanly